port/unix/omrsysinfo.c

```diff
@@ -7783,7 +7783,53 @@ omrsysinfo_get_processes(struct OMRPortLibrary *portLibrary, OMRProcessInfoCallb
 	portLibrary->mem_free_memory(portLibrary, command);
 	closedir(dir);
 	return callback_result;
-#else /* defined(LINUX) */
+#elif defined(OSX) /* defined(LINUX) */
+	uintptr_t callback_result = 0;
+	pid_t *pid_buffer = NULL;
+	int actual_bytes = 0;
+	int i = 0;
+	int num_procs = 0;
+	char path_buf[PROC_PIDPATHINFO_MAXSIZE];
+	int num_bytes = proc_listpids(PROC_ALL_PIDS, 0, NULL, 0);
+	if (num_bytes <= 0) {
+		int32_t rc = findError(errno);
+		portLibrary->error_set_last_error(portLibrary, errno, rc);
+		Trc_PRT_failed_to_call_proc_listpids(rc);
+		return (uintptr_t)(intptr_t)rc;
+	}
+	pid_buffer = (pid_t *)portLibrary->mem_allocate_memory(
+			portLibrary,
+			num_bytes,
+			OMR_GET_CALLSITE(),
+			OMRMEM_CATEGORY_PORT_LIBRARY);
+	if (NULL == pid_buffer) {
+		return OMRPORT_ERROR_SYSINFO_MEMORY_ALLOC_FAILED;
+	}
+	actual_bytes = proc_listpids(PROC_ALL_PIDS, 0, pid_buffer, num_bytes);
+	if (actual_bytes <= 0) {
+		int32_t rc = findError(errno);
+		portLibrary->error_set_last_error(portLibrary, errno, rc);
+		Trc_PRT_failed_to_call_proc_listpids(rc);
+		portLibrary->mem_free_memory(portLibrary, pid_buffer);
+		return (uintptr_t)(intptr_t)rc;
+	}
+	num_procs = actual_bytes / sizeof(pid_t);
+	for (i = 0; i < num_procs; i++) {
+		pid_t pid = pid_buffer[i];
```

Comment:
`path_buf` is only used inside the `for` loop, so it can be declared within the loop’s scope. This limits its visibility and improves readability. Additionally, it can be zero-initialized using `{0}`, which is cleaner and often more efficient than calling `memset()`.
```suggestion
		pid_t pid = pid_buffer[i];
		char path_buf[PROC_PIDPATHINFO_MAXSIZE] = {0};
```

Reply:
Done